lib/browserify-transform.js

```diff
@@ -95,166 +95,31 @@ module.exports = function yoYoify (file, opts) {
-            val += 'arguments[' + argCount + ']'
-            if (src.after) val += ' + ' + JSON.stringify(src.after)
-            resultArgs.push(src.arg)
-            argCount++
-          }
-        })
-      } else {
-        val = JSON.stringify(prop)
-      }
-      if (ksrcs) {
-        k = ''
-        ksrcs.forEach(function (src, index) {
-          if (src.arg) {
-            if (index > 0) val += ' + '
-            if (src.before) val += JSON.stringify(src.before) + ' + '
-            k += 'arguments[' + argCount + ']'
-            if (src.after) k += ' + ' + JSON.stringify(src.after)
-            resultArgs.push(src.arg)
-            argCount++
-          }
-        })
-      } else {
-        k = JSON.stringify(key)
-      }
-      addAttr(elname, k, val)
-    })
-
-    if (Array.isArray(children)) {
-      var childs = []
-      children.forEach(function (child) {
-        var srcs = getSourceParts(child)
-        if (srcs) {
-          var src = srcs[0]
-          if (src.src) {
-            res.push(src.src)
-          }
-          if (src.name) {
-            childs.push(src.name)
-          }
-          if (src.arg) {
-            var argname = 'arguments[' + argCount + ']'
-            resultArgs.push(src.arg)
-            argCount++
-            childs.push(argname)
-          }
-        } else {
-          childs.push(JSON.stringify(child))
-        }
-      })
-      if (childs.length > 0) {
-        needsAc = true
-        res.push('ac(' + elname + ', [' + childs.join(',') + '])')
-      }
-    }
-
-    // Return delim'd parts as a child
-    return DELIM + [elname, res.join('\n'), null].join(DELIM) + DELIM
-  }, { comments: true })
-
-  // Run through hyperx
-  var res = hx.apply(null, args)
-
-  // Pull out the final parts and wrap in a closure with arguments
-  var src = getSourceParts(res)
-  if (src && src[0].src) {
-    var params = resultArgs.join(',')
-
-    node.edit.update('(function () {' +
-      (needsAc ? '\n      var ac = require(\'' + path.resolve(__dirname, 'append-child.js').replace(/\\/g, '\\\\') + '\')' : '') +
-      (needsSa ? '\n      var sa = require(\'' + path.resolve(__dirname, 'set-attribute.js').replace(/\\/g, '\\\\') + '\')' : '') +
-      '\n      ' + src[0].src + '\n      return ' + src[0].name + '\n    }(' + params + '))')
+var apply = transform.factory({
+  arrayExpression: function (elements) {
+    return '[' + elements.join(',') + ']'
+  },
+  objectExpression: function (properties) {
+    return '{' + properties.join(',') + '}'
+  },
+  objectProperty: function (key, value, computed) {
+    return computed
+      ? ('[' + key + ']' + ':' + value)
+      : (key + ':' + value)
+  },
+  stringLiteral: function (value) {
+    return JSON.stringify(value)
+  },
+  callCreateElement (html, tag, props, children) {
+    return html + '.createElement(' + tag + ',' + props + ',' + children + ')'
+  },
+  callObjectAssign (objects) {
+    return 'Object.assign(' + objects.join(',') + ')'
```

Comment:
Object.assign also doesn't exist yet in IE11 … choo uses the `xtend` module, that might be good (deduped = no extra bytes :tada: ). maybe `xtend` can be exposed as `nanohtml/lib/merge` or so, and that can be required in the output, that way we don't add a peer dependency.

IE 11 will probably be dropped in choo 7 but not yet: https://github.com/choojs/choo/pull/616